### PR TITLE
new packages cvise and python-pebble

### PR DESCRIPTION
**Pull request by oltolm**

### CODE REVIEW

**Comment by oltolm** [comment]
I don't know what the problem with mingw64 is.

```

[MSYS2 CI] Processing changes

./.ci/ci-build.sh: line 113: test: mingw-w64-python-pebble: binary operator expected
```

File: mingw-w64-cvise/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "${MINGW_PACKAGE_PREFIX}-llvm>9.0.0"
+)
+checkdepends=("${MINGW_PACKAGE_PREFIX}-python-pytest")
+optdepends=("${MINGW_PACKAGE_PREFIX}-colordiff")
+source=("https://github.com/marxin/cvise/archive/v$pkgver.tar.gz")
+sha256sums=('ee9bdfce6f139d0317c98d9c9b51cc68bcaead059de97aac2cf81d71f2215f54')
+
+prepare() {
+    pip install pebble
+}
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:31]: don't install packages with pip.
It's already in depends().

File: mingw-w64-cvise/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Maintainer: Oleg Tolmatcev <oleg.tolmatcev@gmail.com>
+
+_realname=cvise
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=2.10.0
+pkgrel=1
+pkgdesc="Super-parallel Python port of the C-Reduce"
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:8]:
```suggestion
pkgdesc="Super-parallel Python port of the C-Reduce (mingw-w64)"
```

File: mingw-w64-python-pebble/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Maintainer: Oleg Tolmatcev <oleg.tolmatcev@gmail.com>
+
+_realname=pebble
+pkgbase=mingw-w64-python-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-python-${_realname}")
+pkgver=5.0.7
+pkgrel=1
+pkgdesc="Pebble provides a neat API to manage threads and processes within an application"
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:8]:
```suggestion
pkgdesc="Pebble provides a neat API to manage threads and processes within an application (mingw-w64)"
```

File: mingw-w64-python-pebble/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+depends=("${MINGW_PACKAGE_PREFIX}-python")
+makedepends=("${MINGW_PACKAGE_PREFIX}-python-setuptools")
+optdepends=()
+provides=()
+conflicts=()
+replaces=()
+backup=()
+options=()
+install=
+changelog=
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:26]:
```suggestion
```

File: mingw-w64-python-pebble/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+source=(https://files.pythonhosted.org/packages/source/${_realname:0:1}/$_realname/${_realname^}-$pkgver.tar.gz)
+noextract=()
+sha1sums=('0142bd51e3baa2d7afb6e0723f536d74c4d20bed')
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:29]: use sha256sums

File: mingw-w64-python-pebble/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+optdepends=()
+provides=()
+conflicts=()
+replaces=()
+backup=()
+options=()
+install=
+changelog=
+source=(https://files.pythonhosted.org/packages/source/${_realname:0:1}/$_realname/${_realname^}-$pkgver.tar.gz)
+noextract=()
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:28]:
```suggestion
```

File: mingw-w64-python-pebble/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+package() {
+    cd "$srcdir/${_realname^}-$pkgver"
+    MSYS2_ARG_CONV_EXCL="--prefix=" \
+    ${MINGW_PREFIX}/bin/python setup.py install --prefix=${MINGW_PREFIX} \
+        --root="${pkgdir}" --optimize=1 --skip-build
+
+    install -Dm644 LICENSE "${pkgdir}${MINGW_PREFIX}/share/licenses/python-${_realname}/LICENSE"
+}
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:43]: Please fellow mingw-w64-PKGBUILD-template/PKGBUILD.python.sh

File: mingw-w64-python-pebble/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+pkgrel=1
+pkgdesc="Pebble provides a neat API to manage threads and processes within an application"
+arch=('any')
+mingw_arch=('clang64' 'ucrt64' 'mingw64')
+msys2_references=(
+  'pypi: Pebble'
+)
+url="https://github.com/noxdafox/pebble"
+license=('LGPL')
+groups=()
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:16]:
```suggestion
```

File: mingw-w64-cvise/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    "${MINGW_PACKAGE_PREFIX}-clang>9.0.0"
+    "${MINGW_PACKAGE_PREFIX}-python-chardet"
+    "${MINGW_PACKAGE_PREFIX}-python-pebble"
+    "${MINGW_PACKAGE_PREFIX}-python-psutil"
+    "${MINGW_PACKAGE_PREFIX}-unifdef"
+)
+makedepends=(
+    "${MINGW_PACKAGE_PREFIX}-cc"
+    "${MINGW_PACKAGE_PREFIX}-cmake"
+    "${MINGW_PACKAGE_PREFIX}-llvm>9.0.0"
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:23]:
```suggestion
    "${MINGW_PACKAGE_PREFIX}-llvm"
    "${MINGW_PACKAGE_PREFIX}-clang-tools-extra"
```

File: mingw-w64-cvise/PKGBUILD [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      -Wno-dev
+    cmake --build build
+}
+
+check() {
+    ctest --test-dir build --output-on-failure
+}
+
+package() {
+    cmake --install build --prefix "$pkgdir$MINGW_PREFIX"
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:49]:
```suggestion
  ${MINGW_PREFIX}/bin/cmake \
    -B build-${MSYSTEM} \
    -S "$_realname-$pkgver" \
    -G Ninja \
    -DCMAKE_BUILD_TYPE='Release' \
    -DCMAKE_INSTALL_PREFIX="${MINGW_PREFIX}" \
    -DCMAKE_INSTALL_LIBEXECDIR=lib \
    -Wno-dev

  ${MINGW_PREFIX}/bin/cmake --build build-${MSYSTEM}
}

check() {
    ${MINGW_PREFIX}/bin/ctest --test-dir build --output-on-failure
}

package() {
  ${MINGW_PREFIX}/bin/cmake --install build-${MSYSTEM} --prefix "${pkgdir}${MINGW_PREFIX}"
```

File: mingw-w64-python-pebble/PKGBUILD [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+pkgver=5.0.7
+pkgrel=1
+pkgdesc="Pebble provides a neat API to manage threads and processes within an application (mingw-w64)"
+arch=('any')
+mingw_arch=('clang64' 'ucrt64' 'mingw64')
+msys2_references=(
+  'pypi: Pebble'
+)
+url="https://github.com/noxdafox/pebble"
+license=('LGPL')
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:15]:
```suggestion
license=('spdx:LGPL-3.0-only')
```

File: mingw-w64-cvise/PKGBUILD [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+_realname=cvise
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=2.10.0
+pkgrel=1
+pkgdesc="Super-parallel Python port of the C-Reduce (mingw-w64)"
+arch=('any')
+mingw_arch=('clang64' 'ucrt64' 'mingw64')
+url="https://github.com/marxin/cvise"
+license=('custom:University of Illinois')
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:12]:
```suggestion
license=('spdx:BSD-3-Clause-Clear')
```

File: mingw-w64-python-pebble/PKGBUILD [hunk 13]
```diff
@@ -0,0 +1,57 @@
+# Maintainer: Oleg Tolmatcev <oleg.tolmatcev@gmail.com>
+
+_realname=pebble
+pkgbase=mingw-w64-python-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-python-${_realname}")
+pkgver=5.0.7
+pkgrel=1
+pkgdesc="Pebble provides a neat API to manage threads and processes within an application (mingw-w64)"
+arch=('any')
+mingw_arch=('clang64' 'ucrt64' 'mingw64')
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:10]:
```suggestion
mingw_arch=('clang64' 'clangarm64' 'ucrt64' 'mingw64')
```

File: mingw-w64-cvise/PKGBUILD [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+)
+checkdepends=("${MINGW_PACKAGE_PREFIX}-python-pytest")
+optdepends=("${MINGW_PACKAGE_PREFIX}-colordiff")
+source=("https://github.com/marxin/cvise/archive/v$pkgver.tar.gz")
+sha256sums=('ee9bdfce6f139d0317c98d9c9b51cc68bcaead059de97aac2cf81d71f2215f54')
+
+prepare() {
+    echo nothing to do
+}
+
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:33]:
```suggestion
```

File: mingw-w64-cvise/PKGBUILD [hunk 15]
```diff
@@ -0,0 +1,50 @@
+# Maintainer: Oleg Tolmatcev <oleg.tolmatcev@gmail.com>
+
+_realname=cvise
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=2.10.0
+pkgrel=1
+pkgdesc="Super-parallel Python port of the C-Reduce (mingw-w64)"
+arch=('any')
+mingw_arch=('clang64' 'ucrt64' 'mingw64')
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:10]:
```suggestion
mingw_arch=('clang64' 'clangarm64' 'ucrt64' 'mingw64')
```

**Comment by oltolm** [comment]
I fixed everything.

File: mingw-w64-python-pebble/PKGBUILD [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+  ${MINGW_PREFIX}/bin/python -m build --wheel --skip-dependency-check --no-isolation
+}
+
+check() {
+  msg "Python test for ${MSYSTEM}"
+  cd "${srcdir}/python-build-${MSYSTEM}"
+
+# The test command will usually depend upon what is contained in the tox.ini file
+# or in the [testenv:py] section of the pyproject.toml file.
```
Comment (MehdiChinoune) [inline mingw-w64-python-pebble/PKGBUILD:44]:
```suggestion
```

File: mingw-w64-cvise/PKGBUILD [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+)
+checkdepends=("${MINGW_PACKAGE_PREFIX}-python-pytest")
+optdepends=("${MINGW_PACKAGE_PREFIX}-colordiff")
+source=("https://github.com/marxin/cvise/archive/v$pkgver.tar.gz")
+sha256sums=('ee9bdfce6f139d0317c98d9c9b51cc68bcaead059de97aac2cf81d71f2215f54')
+
+build() {
+    cmake -B build -S "$_realname-$pkgver" \
+      -G Ninja \
+      -DCMAKE_BUILD_TYPE='Release' \
```
Comment (MehdiChinoune) [inline mingw-w64-cvise/PKGBUILD:34]:
```suggestion
```

**Comment by oltolm** [comment]
Fixed again.

**Pull request merged**